cloudinit/sources/DataSourceAeza.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (C) 2024 Aeza.net.
+#
+# Author: Egor Ternovoy <cofob@riseup.net>
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+from cloudinit import sources, util
+import cloudinit.sources.helpers.aeza as aeza_helper
+
+BASE_URL_V1 = "http://77.221.156.49/v1/cloudinit"
+
+BUILTIN_DS_CONFIG = {
+    "retries": 60,
+    "timeout": 2,
+    "wait_retry": 2,
+    "metadata_url": BASE_URL_V1 + "/{id}/meta-data",
+    "userdata_url": BASE_URL_V1 + "/{id}/user-data",
+    "vendordata_url": BASE_URL_V1 + "/{id}/vendor-data",
+}
+
+class DataSourceAeza(sources.DataSource):
+
+    dsname = "Aeza"
+
+    def __init__(self, sys_cfg, distro, paths, ud_proc=None):
+        sources.DataSource.__init__(self, sys_cfg, distro, paths, ud_proc)
+        self.metadata = {}
+        self.vendordata_raw = None
+        self.userdata_raw = ""
+
+        self.ds_cfg = util.mergemanydict(
+            [
+                util.get_cfg_by_path(sys_cfg, ["datasource", "Aeza"], {}),
+                BUILTIN_DS_CONFIG,
+            ]
+        )
+
+        self.timeout = self.ds_cfg["timeout"]
+        self.wait_retry = self.ds_cfg["wait_retry"]
+        self.retries = self.ds_cfg["retries"]
+
+        self.metadata_address = self.ds_cfg["metadata_url"]
+        self.userdata_address = self.ds_cfg["userdata_url"]
+        self.vendordata_address = self.ds_cfg["vendordata_url"]
+
+    def _get_data(self):
```

Comment:
we need a ds_detect method on this class which returns True/False that cloud-init python code queries first to see if this platform is even detectable as Azea before trying to read metadata. We don't want to waste time trying to reach out to a remote IP on systems if they are unlikely to be Aeza as it costs boot time for every environment which is not aeza. Some images do not call ds-identify early in boot, and those systems have to walk through all python datasources to run _get_data which ends up taking 10+ minutes in some cases due to network retries and timeouts if datasources don't have an early exit via ds_detect() which is called before _get_data

Reply:
Fixed in 4350dc221feb40ba2e277706e890126f92cee282

Reply:
Looks good thank you.__